hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSourceStorage.scala

```diff
@@ -284,6 +289,120 @@ class TestCOWDataSourceStorage extends SparkClientFunctionalTestHarness {
       .filter(action => action.equals(HoodieTimeline.REPLACE_COMMIT_ACTION)).size > 0)
   }
 
+  @ParameterizedTest
+  @CsvSource(value = Array(
+    "org.apache.hudi.client.validator.SqlQueryEqualityPreCommitValidator|"
+      + "select count(*) from <TABLE_NAME> where driver is null|true|false|false",
+    "org.apache.hudi.client.validator.SqlQueryEqualityPreCommitValidator|"
+      + "select count(*) from <TABLE_NAME> where driver is null|true|true|false",
+    "org.apache.hudi.client.validator.SqlQueryEqualityPreCommitValidator|"
+      + "select count(*) from <TABLE_NAME> where rider is null|true|true|true",
+    "org.apache.hudi.client.validator.SqlQueryEqualityPreCommitValidator|"
```

Comment:
Can we avoid to hardcode these validators?

Reply:
Addressed.